### DISCUSSION TITLE
Getting datespanValue

### DISCUSSION THREAD

**Discussion by leotumwattana** [soulverteam/SoulverCore#8] [Q&A] [answered]
Hi @zcohan,

I'm currently evaluating `SoulverCore` for a project of mine, but I'm not getting the `datespanValue` result I was expecting.

Maybe you can point me to what I may be doing wrong?

```swift 

var customization = EngineCustomization.standard

customization.featureFlags.seeksFutureDate = true

customization.featureFlags.calendarCalculations = true

let calculator = Calculator(customization: customization)

let values = calculator.dateFor("3-4 march") // As per sample given in code base

let dateSpan = values?.result.evaluationResult.datespanValue 

print(dateSpan) // => nil 

```

Thanks,

Leo

**Comment by zcohan** [comment]
Hi @leotumwattana,

The `dateFor` function on `Calculator` only returns date results at present. In the next build I'll add support for datespans as well if you need this.

As a workaround in the meantime you can also get the datespan from the parsed expression:
`let datespan = calculator.calculate("March 12 - 14").parsedExpression?.firstToken(ofType: .datespan)?.datespanValue`

**Comment by leotumwattana** [comment]
Hi @zcohan,

Thanks for your quick reply. Adding datespan to `dateFor` would be an API enhancement. 

However, it seems there are a few cases where the datespan parsing is having some issues:

```swift
let datespan = calculator.calculate("March 3 5pm-6pm").parsedExpression?.firstToken(ofType: .datespan)?.datespanValue

// Result: "Mar 3, 2005 at 12:00 PM" - "Mar 6, 2005 at 12:00 PM"
// Expected: "Mar 3, 2024 5pm" - "Mar 3, 2024 6pm"
```

```swift
let datespan = calculator.calculate("March 3 5pm for 2 hours").parsedExpression?.firstToken(ofType: .datespan)?.datespanValue

// Result: nil
// Expected: "Mar 3, 2024 5pm" - "Mar 3, 2024 7pm"
```

**Reply by zcohan** [reply]
A datespan in SoulverCore is a span of dates in the format `March 12-14`. Other dates & times are generally parsed as datestamps.

For example, your first expression has two datestamps (`March 3 5pm` and `6pm [today]`), your second expression contains a datestamp (`March 3 5pm` and a unit expression of time (`2 hours`).

**Reply by leotumwattana** [reply]
I see. That technically makes sense, but for the use case of `natural language date input ... (similar to ... Fantastical)` that would not work very well.

For example, typing "Monday 5pm to 7pm" in Fantasical would result in a date interval of `startDate: Nov 18th 2024 5pm, endDate: Nov 18th 2024 7pm`

Would you consider that as outside the scope of `SolverCore`? 

If so, then I probably need to consider other solutions. 

![IMG_9689](https://github.com/user-attachments/assets/07db5ba3-279e-475f-9547-fa3218d66160)

**Comment by zcohan** [comment, marked as answer]
Try the new `dateIntervalFor(_ expression: String) -> DateInterval?` on `Calculator` in version `3.0.0`.
